integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/scaletest/QuerySpecs.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tests.scaletest
+
+import scala.io.Source
+
+import com.nvidia.spark.rapids.tests.scaletest.ScaleTest.Config
+
+import org.apache.spark.sql.SparkSession
+
+class QuerySpecs(config: Config, spark: SparkSession) {
+  private val baseInputPath = s"${config.inputDir}/" +
+    "SCALE_" +
+    s"${config.scaleFactor}_" +
+    s"${config.complexity}_" +
+    s"${config.format}_" +
+    s"${config.version}_" +
+    s"${config.seed}"
+  private val tables = Seq("a_facts",
+    "b_data",
+    "c_data",
+    "d_data",
+    "e_data",
+    "f_facts",
+    "g_data")
+
+  /**
+   *  Read data and initialize temp views in Spark
+   */
+  def initViews(): Unit = {
+
+    for (table <- tables) {
+      spark.read.format(config.format).load(s"$baseInputPath/$table")
+        .createOrReplaceTempView(table)
+    }
+  }
+
+  /**
+   * Parse one query template to get the real query string that Spark SQL accepts
+   * @param queryTemplate template query
+   * @return Spark SQL compatible query
+   */
+  private def parseQueryTemplate(queryTemplate: String): String = {
```

Comment:
How do you handle things like with query 22?

```
SELECT b_key3_*, complexity number of aggregations that are SUMs of 2 or more numeric data columns multiplied together or MIN/MAX of any data column FROM b_data GROUP BY b_key3_*.
```

I am not sure that extending SQL with templates is really what we want to do. I don't see a lot of value in doing this over something like

```
def genQuery(query: String, scale: Int, complexity: Int): String = query match {
  case "q1" =>
    s"SELECT a_facts.*, ${expandColumns("b_data_", 1, 10)} FROM b_data JOIN a_facts WHERE primary_a = b_foreign_a"
  ...
}
```

To me the text template adds in complexity that not only could have bugs in it, but is something we have to document also.  I am fine with having a tool that will generate the SQL and write it out to a file and a separate tool that will take the SQL and run it. I am even fine if the tool will generate the SQL on the fly for us to run. But I am not a fan of the templated queries.

Reply:
I see...It looks like query 22 represents another type of query template which is typically "human language" not simply a fixed template format. It's really hard to handle.
So I plan to put all queries into the code directly for now. it's a fixed way but with less issues.